packages/notte-core/src/notte_core/credentials/base.py

```
@@ -645,16 +650,17 @@ async def replace_credentials(
-
-        for action_key, original_action_value in values_to_replace.items():
-            placeholder_value = get_str_value(original_action_value)
+        match action:
+            case FillAction() | MultiFactorFillAction() | FallbackFillAction():
+                values_to_replace = {"key": action.value}
+            case FormFillAction():
+                values_to_replace = action.value
+            case SelectDropdownOptionAction():
+                values_to_replace = {"key": action.value}
+            case _:
+                raise ValueError(f"Cant put credentials for action type {type(action)}")
```

Ruff (0.12.2)

Avoid specifying long messages outside the exception class

(TRY003)